autoload.js

```diff
@@ -59,87 +141,173 @@ if (typeof FileReader !== "undefined") {
-          writable: false,
-          value: Array.prototype.concat.apply([], labels)
-        })
-        return this.labels
+        return Array.prototype.concat.apply([], labels)
       }
     })
 
+    /**
+     * Define a lazily evaluated memonized property on the provided object with
+     * the provided key using the provided property descriptor.
+     *
+     * It has the same parameters as `Object.defineProperty`.
+     *
+     * @param {Object} object
+     * @param {String} key
+     * @param {Object} descriptor
+     */
+    function _defineLazyProperty(object, key, descriptor) {
+      var getter = descriptor.get,
+          writable = descriptor.writeable || false
+
+      delete descriptor.writeable
+      descriptor.get = function init() {
+        delete descriptor.get
+        delete descriptor.set
+        descriptor.value = getter.call(this)
+        descriptor.writeable = writable
+        Object.defineProperty(this, key, descriptor)
+        return descriptor.value
+      }
+      Object.defineProperty(object, key, descriptor)
+    }
+
+    /**
+     * Creates and returns a dispatcher function bound to the input element
+     * this is attached to.
+     *
+     * @param {File} file to set as relatedTarget on the dispatached event
+     * @return {dispatchEventAsync} dispatcher function
+     */
     AutoFileReader.prototype.dispatcher = function AutoFileReader$dispatcher(file) {
-      var input = this.target
+      var dispatchEvent = this.target.dispatchEvent.bind(this.target)
 
-      return function dispatchEvent(event) {
+      /**
+       * @callback dispatchEventAsync
+       * @param {Event} event to dispatch asyncronously
+       */
+      return function dispatchEventAsync(event) {
         event.relatedTarget = file
 
         // Dispatch asyncronous to avoid "The event is already being dispatched"
-        setTimeout(function dispatchAsync() {
-          input.dispatchEvent(event)
-        }, 0)
+        setTimeout(dispatchEvent, 0, event)
       }
     }
 
+    /**
+     * Read the provided file using a FileReader and pipe the readers events to
+     * the input element this is attached to.
+     *
+     * @param {File} file to read
+     */
     AutoFileReader.prototype.read = function AutoFileReader$read(file) {
-      var reader = file.reader = new FileReader(),
-          dispatchEvent = this.dispatcher(file)
+      var dispatchEvent = this.dispatcher(file)
+
+      if (file.size < this.maxSize) {
+        file.reader = new FileReader()
+
+        // Pipe events from the Filereader to this input element, but use the
+        // "on<event>" attributes to avoid memoery leaks.
+        Object.keys(file.reader).forEach(function attachDispatcher(name) {
+          if (name.slice(0, 2) === "on") {
+            file.reader[name] = dispatchEvent
+          }
+        })
+
+        // Start reading
+        file.reader[this.format](file)
+      } else {
+        console.warn.apply(console,
```

Comment:
Yes, since I use a `concat` on the line below to flatten the return values from `_humansize`. It is just for debugging though, I can change it or just remove it.